apps/gustaf/gustaf_matepairs.h

```diff
@@ -99,15 +99,28 @@ _importSequences(CharString const & fileNameL,
         append(qual, qualR);
         appendValue(seqs, seq, Generous());
         appendValue(quals, qual, Generous());
-        appendValue(ids, id, Generous());
+       // appendValue(ids, id, Generous());
 
         _getShortId(sId, id);
-        if (!_checkUniqueId(sId, id, ids, sIds))
-            ++counter;
-        appendValue(sIds, sId);
+        appendValue(ids, sId, Generous());
         clear(seq);
         clear(qual);
     }
+    
+    // Check for dupliacte id entries.
+    StringSet<TId> uniqueIds = ids;
+    std::sort(begin(uniqueIds, Standard()), end(uniqueIds, Standard()), IdComparator<TId>());  // O(n*log(n))
+    TIdSetIterator itOldEnd = end(uniqueIds, Standard());
+    TIdSetIterator itNewEnd = std::unique(begin(uniqueIds, Standard()), end(uniqueIds, Standard()), IdComparator<TId>());  // O(n)
+
+    --itNewEnd;
```

Comment:
I don't know what it does, but it does not work the way we thought. In your version, itNewEnd != itOldEnd was always the case even if there are no nonunique IDs. Also, I couldn't get the number of nonunique IDs right, it was always wrong if there was more than one nonunique ID (especially for different nonunique IDs). I had enough after more than 2 hours, you are welcome to play around with this.